graphql-language/src/ast.rs

```diff
@@ -7,22 +7,23 @@ use source::Source;
 
 /// Contains some character offsets that identify where the source of the AST
 /// is from.
+#[derive(Clone, Copy)]
 pub struct Location<'a> {
   pub start: usize,
   pub end: usize,
-  pub source: Option<&'a Source<'a>>
+  pub source: Source<'a>
 }
 
 /// All AST node types implement this trait.
 pub trait Node<'a> {
-  fn location(&self) -> Option<&Location<'a>>;
```

Comment:
In testing we can have a special impl for `Location` that just returns a dummy loc.